OutOfSchool/OutOfSchool.Common/Models/Either.cs

```diff
@@ -29,21 +29,66 @@ private interface IEither
 
     public static implicit operator Either<TL, TR>(TR right) => CreateRight(right);
 
+    /// <summary>
+    /// If left value is assigned, execute an action on it.
+    /// </summary>
+    /// <param name="leftAction">Action to execute.</param>
+    public void DoLeft(Action<TL> leftAction)
+    {
+        ArgumentNullException.ThrowIfNull(leftAction);
+
+        if (imp is Left left)
+        {
+            leftAction(left.Value);
+        }
+    }
+
+    /// <summary>
+    /// Tries to get the Left value.
+    /// </summary>
+    /// <param name="value">The Left value if present.</param>
+    /// <returns>true if the Either is Left; otherwise, false.</returns>
+    public bool TryGetLeft(out TL value)
```

Comment:
I'm bit opinionated, but don't like this `TryGetLeft`/`Right` approach. 
My reason - it counters to the whole idea of Either monad that is accessible only through limited set of api methods.
Yes, this `TryGet...` provides more flexible approach, but for me this flexibility has same nature as dynamic typing vs static typing.